### PR TITLE
test: unify some tests into the check frame

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?
Currently, TestInfo, TestSyncersimple, TestSingle, and TestCluster use the original go test frame, it's better to use our check frame. 

### What is changed and how it works?

The problem that we didn't use the check frame before is that the function `integration.NewClusterV3` accepts `*testing.T`, since we don't want to test `etcd` itself, we can pass `nil` into it directly. Then we can use our check frame. 

### Related changes
- Need to cherry-pick to the release branch

### Check List 

Tests 

- No code

Side effects

### CODE REVIEW

File: domain/domain_test.go [hunk 1]
```diff
@@ -48,11 +48,19 @@ func TestT(t *testing.T) {
...
 }
 
-var _ = Suite(&testSuite{})
+var _ = SerialSuites(&testSuite{})
 
 type testSuite struct {
 }
 
+func (s *testSuite) SetUpSuite(c *C) {
+	testleak.BeforeTest()
```
Comment (zimulala) [inline domain/domain_test.go:57]: Could we put it to `TestT`? Then we can remove `SetUpSuite` and `TearDownSuite`.

Comment (wjhuang2016) [reply domain/domain_test.go:57]: If we put it into `TestT`, it will check only one time, not per `testSuite`.

Comment (zimulala) [reply domain/domain_test.go:57]: I feel no problem. It shouldn't matter if you check once or twice. After all tests are completed, check once for any extra goroutine.

**Pull request closed without merging**